spec/models/dynamic_field_group_spec.rb

```diff
@@ -0,0 +1,339 @@
+          FactoryBot.create(:dynamic_field, string_key: 'role', dynamic_field_group: parent)
+          parent.reload
+        end
+
+        it 'does not save' do
+          expect(dynamic_field_group.save).to be false
+        end
+
+        it 'returns correct error' do
+          dynamic_field_group.save
+          expect(dynamic_field_group.errors.full_messages).to include 'String key has already been taken'
+        end
+      end
+
+      context 'and the sibiling is a dynamic_field_group' do
+        let(:parent) { FactoryBot.create(:dynamic_field_category) }
+        let(:dynamic_field_group) { FactoryBot.build(:dynamic_field_group, parent: parent) }
+
+        before do
+          FactoryBot.create(:dynamic_field_group, string_key: 'name', parent: parent)
+          parent.reload
+        end
+
+        it 'does not save' do
+          expect(dynamic_field_group.save).to be false
+        end
+
+        it 'returns correct error' do
+          dynamic_field_group.save
+          expect(dynamic_field_group.errors.full_messages).to include 'String key has already been taken'
+        end
+      end
+    end
+
+    context 'when creating a group that shares the same string_key as a non-sibling group' do
+      let(:dynamic_field_group) { FactoryBot.create(:dynamic_field_group) }
+      let(:dynamic_field) { FactoryBot.build(:dynamic_field, string_key: 'name', dynamic_field_group: dynamic_field_group) }
+
+      it 'saves the object' do
+        expect(dynamic_field_group.save).to be true
+      end
+    end
+  end
+
+  describe '#parent' do
+    let(:dynamic_field_group) { FactoryBot.build(:dynamic_field_group, :child) }
+
+    before do
+      dynamic_field_group.parent = parent
+      dynamic_field_group.save
+      parent.reload
+    end
+
+    context 'when setting DynamicFieldCategory as parent' do
+      let(:parent) { FactoryBot.create(:dynamic_field_category) }
+
+      it 'adds parent' do
+        expect(dynamic_field_group.parent).to eq parent
+      end
+
+      it 'adds dynamic field group to dynamic field category' do
+        expect(parent.dynamic_field_groups).to match_array [dynamic_field_group]
+      end
+    end
+
+    context 'when setting DynamicFieldGroup as parent' do
+      let(:parent) { FactoryBot.create(:dynamic_field_group) }
+
+      it 'adds parent' do
+        expect(dynamic_field_group.parent).to eq parent
+      end
+
+      it 'adds dynamic field group to parent dynamic field group' do
+        expect(parent.dynamic_field_groups).to match_array [dynamic_field_group]
+      end
+    end
+
+    context 'when setting parent to invalid parent type' do
+      let(:parent) { FactoryBot.create(:user, email: 'random_user@example.com') }
+
+      it 'does save' do
+        expect(dynamic_field_group.save).to be false
+      end
+
+      it 'returns correct error' do
+        dynamic_field_group.save
+        expect(dynamic_field_group.errors.full_messages).to include 'Parent type is not included in the list'
```

Comment:
```suggestion
        expect(dynamic_field_group.errors.full_messages).to include 'Invalid parent type'
```

Reply:
The reason I suggested this change is that it seems confusing for a user of the API or the GUI to see a message that refers to "the list".  What list should they assume the error is referring to?

Reply:
I get that. Like maybe "not included in valid list" or something like that. But again, i feel like we should just change this across the app. I don't know how to do that, yet, or if its even possible

Reply:
We might be able to use localization for this, which might be awesome: https://github.com/rails/rails/blob/e69ff43060c1194d2a3bd9b8d9e23f3ae26b84b5/activemodel/lib/active_model/errors.rb#L401

Reply:
Gotta love rails: https://guides.rubyonrails.org/i18n.html#error-message-scopes

Reply:
@elohanlon Did you want to go through and change the error messages for list validation and belongs to validations across the app? And if so, what should we change the wording to?

Reply:
As discussed we change the belongs_to validation to 'is_required' and the inclusion validation to 'is not among the list of allowed values'